.spelling

```diff
@@ -0,0 +1,147 @@
+# markdown-spellcheck spelling configuration file
+# Format - lines beginning # are comments
+# global dictionary is at the start, file overrides afterwards
+# one word per line, to define a file override use ' - filename'
+# where filename is relative to this configuration file
+AlertManager
+Ansible
+CentOS
+CloudFront
+CockroachDB
+Conda
+CoreOS
+Couchbase
+Crunchbase
+D2iQ
+Datastax
+Dex
+Dispatchfile
+Dockerfile
+ESXi
+ElasticSearch
+Flagger
+Flink
+GitOps
+Github
+Golang
+Grafana
+HAProxy
+Hadoop
+Hashicorp
+Horovod
+Iinc
+InfluxData
+Istio
+Jupyter
+JupyterHub
+JupyterLab
+K8s
+Katib
+Kerberos
+Kibana
+Knative
+Kommander
+Konvoy
+Konvoy
+Kubeflow
+Kubernetes
+Linkerd
+Logstash
+MXNet
+MacBook
+MacOS
+Mesos
+MongoDB
+MySQL
+Nagios
+Nginx
+NiFi
+OneLogin
+ReLU
+OpenID
+OpenResty
+Percona
+PerconaXtraDB
+PipelineRuns
+PostgreSQL
+PyTorch
+Segment.io
+Splunk
+Tekton
+Telegraf
+TensorFlow
+TitanDB
+Toree
+VMWare
+VMware
+VirtualService
+XtraDB
+addon
+artifact
+artifacts
+autoscaler
+config
+configs
+hostname
+kubeaddon
+kubeaddons
+lifecycle
+mTLS
+plugin
+spec
+hyperparameter
+hyperparameters
+serverless
+SQLite
+KFServing
+MLOps
+MinIO
+MutatingWebhookConfiguration
+base64
+gRPC
+
+#
+# Stuff needed to make Kubeflow error free
```

Comment:
i like your idea of having an alphabetically sorted list so that it's easier to go through and fix. I personally like having files in subdirs for each product since that means they can all be sorted alphabetically but will leave it up to you to choose a global spelling file or not.

Reply:
i'd actually leave that to @cmastr. i think that starting with a single global file brings most value as that enables us to spot/fix existing typos most easily.

i also think that having it completely sorted alphabetically is the best idea here. this PR is rather a demo of how things might work and that we build upon later.